addons/account/models/account_move.py

```diff
@@ -4409,6 +4419,57 @@ def _compute_tax_line_id(self):
             # A constraint on account.tax.repartition.line ensures both those fields are mutually exclusive
             record.tax_line_id = rep_line.invoice_tax_id or rep_line.refund_tax_id
 
+    @api.depends('tax_line_id.tax_exigibility', 'tax_ids.tax_exigibility')
+    def _compute_all_taxes_are_on_payment(self):
+        for record in self:
+            record.has_taxes = (record.tax_line_id or record.tax_ids)
+            record.all_taxes_are_on_payment = (
+                (not record.tax_line_id or record.tax_line_id.tax_exigibility == "on_payment")
+                and all(tax.tax_exigibility == "on_payment" for tax in record.tax_ids)
+            )
+
+    @api.depends('tax_ids')
+    def _compute_tax_move_line_ids(self):
+        # prefetch
+        self.mapped('move_id.line_ids')
+        def endswith(a, b):
+            for i, x in enumerate(reversed(a)):
+                if b[-1-i] != x:
+                    return False
+            return True
+
+        for base_line in self:
+            tax_line_ids = []
+            if not base_line.tax_repartition_line_id:
+                tax_line_ids = base_line.move_id.line_ids.filtered(
+                    lambda tax_line:
+                        tax_line.tax_line_id
+                        and tax_line.tax_repartition_line_id
+                        and (tax_line.group_tax_id or tax_line.tax_line_id) in base_line.tax_ids  # TODO: is it needed?
+                        and base_line.partner_id == tax_line.partner_id  # TODO: is it needed?
+                        and base_line.currency_id == tax_line.currency_id  # TODO: is it needed?
```

Comment:
Regarding the TODO, these are needed for Journal Entries, where you can edit those values.